tools/mo/openvino/tools/mo/serialize.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+from openvino.tools.mo.back.ie_ir_ver_2.emitter import append_ir_info
+from openvino.tools.mo.utils.cli_parser import get_meta_info
+from openvino.tools.mo.utils.error import Error
+from openvino.tools.mo.utils.telemetry_utils import get_tid
+from openvino.tools.mo.utils.utils import refer_to_faq_msg
+from openvino.tools.mo.utils.version import get_simplified_mo_version
+
+try:
+    import openvino_telemetry as tm
+except ImportError:
+    import openvino.tools.mo.utils.telemetry_stub as tm
+
+
+def serialize(ngraph_function, xml_path, argv=None):
+    """
+    Serializes nGraph function to IR.
+    :param ngraph_function: nGraph function to serialze.
+    :xml_path: path to IR.
+    :argv: Command line arguments info which will be appended to IR.
+    """
+    from openvino.runtime import serialize  # pylint: disable=import-error,no-name-in-module
+    telemetry = tm.Telemetry(tid=get_tid(), app_name='Model Optimizer', app_version=get_simplified_mo_version())
+    telemetry.start_session('mo')
+    telemetry.send_event('mo', 'version', get_simplified_mo_version())
+
+    output_dir = os.path.dirname(xml_path)
+    if not os.path.exists(output_dir):
+        try:
+            os.makedirs(output_dir)
+        except PermissionError as e:
+            raise Error("Failed to create directory {}. Permission denied! " +
+                        refer_to_faq_msg(22),
+                        output_dir) from e
+    else:
+        if not os.access(output_dir, os.W_OK):
+            raise Error("Output directory {} is not writable for current user. " +
+                        refer_to_faq_msg(22), output_dir)
+
+    serialize(ngraph_function, xml_path.encode('utf-8'), xml_path.replace('.xml', '.bin').encode('utf-8'))
+
+    if argv is not None:
```

Comment:
`argv` is not something that should be exposed to the user, it is an internal detail how MO as cmdline tool is designed. So it shouldn't be passed to this function. So no extra info serialized at least passed as `argv`. If we really want to still embed cmdline-like info to IR, then it should be passed in rt_info in function itself and restored here. And this still supports the idea to get rid off `mo.serialize`. But still we have that extra functionality with output_dir creation. Can we move this to ov.serialize?

Reply:
I made a ticket for adding MO meta data to rt_info: 88606. In this PR I added warning that MO temporary doesn't generate metadata.

Reply:
I believe it is not what we discussed. We discussed that temporary mo.convert as API temporary won't embed meta info, but nothing was said about mo as a tool. Tool should continue generating meta-info. So you have to embed this meta info somehow for now. If it is too hard, let's re-iterate on that.